### DISCUSSION TITLE
Good time to start contributing to taskwarrior?

### DISCUSSION THREAD

**Discussion by dahelb** [GothenburgBitFactory/taskwarrior#3356] [General]
Hey there! I'm eager to begin contributing to Taskwarrior. I have mainly experience with Rust, none with C++.
I've taken a look at the contribution and development guidelines, but I find them a bit scattered, possibly due to the transition of the backend to Taskchampion.

A couple of things are holding me back from contributing:

1. The "good first issue" label appears to be neglected. Many of the listed issues are five years old or older. Should we consider revitalizing this label?
2. It's unclear which part of the project each issue relates to (e.g., CLI or backend), and whether my Rust skills would be beneficial. Could we perhaps use labels to categorize and triage these issues accordingly?

I'm wondering if now is the right time for me to get involved as a new contributor, or if it might be better to wait until version 3 has stabilized further. If the former, I would appreciate some pointers to "good first issues" :)

**Comment by djmitche** [comment]
Hi @davidhelbig! You're right, things are a bit scattered right now during the transition. 

You're right about the "good first issue" -- I just removed it from most of those issues, since the folks who might be able to mentor those issues are no longer around. In general, we'll need to do some serious work triaging issues in the next month or two.

We will be splitting Taskchampion out into a separate repo (#3209), which should help the distinction somewhat. There will still be some Rust in this repo, where it interfaces with Taskchampion, but much less of it.  We've already split out `taskchampion-sync-server`, and in https://github.com/GothenburgBitFactory/taskchampion-sync-server/issues/2 @jcgruenhage suggested switching from actix-web to axum. Maybe that's a place you could chip in? In general I think the sync server needs one or more people to kind of set out a vision for it and then build toward that vision.

Right now, the main work falls into a few categories:
 * Bugfixes for Taskwarrior-3.0 (mostly C++ stuff)
 * Admin (publishing docs in the right places, release process, etc.)
 * More significant refactors for Taskchampion (#2928 and #3058)
 * Taskchampion-sync-server work (needs some planning)
 * New cloud backends

All of those are open to contributors, and the last three focus on Rust but maybe not great "first bugs" in the sense that they're not well-defined and mentored.

What do you think

**Reply by dahelb** [reply]
Thanks for the overview @djmitche !
To be honest, I was hoping to get my feet wet with some more-or-less clearly defined fix, but I will definitely have a look at the things you mentioned. The sync server sounds interesting, I will check if I can make some useful contributions there!

**Reply by djmitche** [reply]
Sounds good! If work on the server doesn't pan out, do consider the cloud backends. I think an Azure backend would be a pretty straightforward port of the GCP backend, once you've found and understood an Azure blob-storage library. AWS is probably a little more complex. There's also the option to build some backends to other storage tools like MongoDB or Couchbase.

**Reply by djmitche** [reply]
I've just created https://github.com/gothenburgBitFactory/taskchampion for Taskchampion development, if you'd like to pitch in there.
